tests_e2e/tests/lib/shell.py

```diff
@@ -46,9 +46,17 @@ def run_command(command: Any, shell=False) -> str:
     NOTE: The command's stdout and stderr are read as text streams.
     """
     process = Popen(command, stdout=PIPE, stderr=PIPE, shell=shell, text=True)
+    timer = threading.Timer(15 * 60, process.kill)  # Kill process after timeout
```

Comment:
this is applied to all commands, not only ssh

also, for ssh, we should terminate the remote process, not the local process

Reply:
yes, I would think we even don't want the commands to stuck. 

is not kill child process which includes remote process? In my experiment processes in the vm also terminated.

Reply:
ah, ok, as long as the remote process is killed too, we are ok